### PR TITLE
functional correctness of hash-to-field

### PR DESCRIPTION
**Pull request by nano-o**

### CODE REVIEW

**Review by andreistefanescu — APPROVED** [review]
lgtm!

**Comment by nano-o** [comment]
This needs to be updated: changing Fp representation breaks it (does not even type-check).

**Comment by msaaltink** [comment]
Probably this just needs to have some `to_Fp` and `from_Fp` sprinkled in places you've mentioned specific field constants.  I can have a look after I finish up what I'm doing now.

File: proof/hash_to_field.saw [hunk 1]
```diff
@@ -15,31 +28,158 @@ let expand_message_spec len_in_bytes aug_len msg_len DST_len = do {
...
+vec_abs_rep_thm <- prove_cryptol {{ \x -> vec384_abs (vec384_rep x) == x}} [];
+
+// A theorem to get rid of map (TODO: does not seem useful so far):
+let map_size_1_term = parse_core "\\(f : Integer -> Vec 6 (Vec 64 Bool)) -> \\(y : Vec 1 Integer) -> map Integer (Vec 6 (Vec 64 Bool)) (\\(x : Integer) -> f x) 1 y";
+let map_size_1_eq_term = beta_reduce_term {{ \x -> map_size_1_term fp_rep x == [fp_rep (x@0)]}};
+map_size_1_thm <- prove_cryptol map_size_1_eq_term [];
+
+// A rewrite to get rid of a triviality (not really necessary):
+let intLe_sym_term = parse_core "\\(x : Integer) -> intLe x x";
+intLe_sym_thm <- really_prove_cryptol (beta_reduce_term {{\x -> intLe_sym_term x == True}}) [];
```
Comment (msaaltink) [inline proof/hash_to_field.saw:123]: Should drop the "really_", here and a few lines down. (line 134)

**Pull request merged**